src/Test/DocTest/Internal/Interpreter.hs

```diff
@@ -27,6 +30,19 @@ import Test.DocTest.Internal.Logging (DebugLogger)
 -- >>> import Test.DocTest.Internal.GhciWrapper (eval)
 -- >>> import Test.DocTest.Internal.Logging (noLogger)
 
+#if __GLASGOW_HASKELL__ <= 906
+-- | If the list is empty returns 'Nothing', otherwise returns the 'init' and the 'last'.
+--
+-- > unsnoc "test" == Just ("tes",'t')
+-- > unsnoc ""     == Nothing
+-- > \xs -> unsnoc xs == if null xs then Nothing else Just (init xs, last xs)
+unsnoc :: [a] -> Maybe ([a], a)
+unsnoc [] = Nothing
+unsnoc [x] = Just ([], x)
+unsnoc (x:xs) = Just (x:a, b)
+    where Just (a,b) = unsnoc xs
```

Comment:
This formulation avoids warnings and unnecessary matching:
```haskell
unsnoc :: [a] -> Maybe ([a], a)
unsnoc = \case
  []   -> Nothing
  x:xs -> Just $ unsnoc1 x xs

unsnoc1 :: a -> [a] -> ([a], a)
unsnoc1 x = \case
  []   -> ([], x)
  y:ys -> first (x:) $ unsnoc1 y ys
```

Reply:
ah thanks!